feature/funding/googleplay/src/main/kotlin/app/k9mail/feature/funding/googleplay/ui/reminder/FundingReminder.kt

```diff
@@ -0,0 +1,71 @@
+package app.k9mail.feature.funding.googleplay.ui.reminder
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Handler
+import android.os.Looper
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import app.k9mail.feature.funding.api.FundingRoute
+import app.k9mail.feature.funding.api.FundingSettings
+import app.k9mail.feature.funding.googleplay.R
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+
+class FundingReminder(
+    private val fundingSettings: FundingSettings,
+) : FundingReminderContract.Reminder {
+
+    private val handler = Handler(Looper.getMainLooper())
+    private var showDialogRunnable: Runnable? = null
+
+    override fun registerReminder(activity: AppCompatActivity, launcherIntent: Intent) {
+        if (!shouldShowFundingReminder()) {
+            return
+        }
+
+        showDialogRunnable = createShowDialogRunnable(activity, launcherIntent)
```

Comment:
This should be a local variable. That way we can get rid of the `!!` and we don't leak the `Activity` when ´FundingReminder` holds on to it via the `showDialogRunnable` property.